### PR TITLE
remove support for generators

### LINKED ISSUE #1021: remove generators
Remove generator snippet interface in favor of async-await and remove `setDefinitionFunctionWrapper` since that was used to turn generator functions into promises. Appears `setDefinitionFunctionWrapper` has been used for other things so need to ensure #997 can address the use case of being able to wrap a step with custom logic.

### PR DESCRIPTION
**Pull request by charlierudolph**
Don't need generators anymore given async / await has been out for a long time.

Initially held off to allow until we had BeforeStep / AfterStep hooks as the definition function wrapper had been used for things like that

Also removes bluebird (between lodash / bluebird - removing 20% of our bundle: https://bundlephobia.com/package/@cucumber/cucumber@7.3.0)

closes #1021

### CODE REVIEW

**Comment by davidjgoss** [comment]
Are we missing commits that actually remove `setDefinitionFunctionWrapper` and the related code in `UserCodeRunner` etc? I think we could also drpo the `unwrappedCode` field.

**Comment by charlierudolph** [comment]
Haha focused too much on bluebird there. Thanks @davidjgoss. Updated

File: src/time_spec.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { describe, it } from 'mocha'
 import { expect } from 'chai'
 import { wrapPromiseWithTimeout } from './time'
 
-describe.only('wrapPromiseWithTimeout()', () => {
```
Comment (charlierudolph) [inline src/time_spec.ts:5]: Had accidentally added this in a previous PR, added the mocha options forbid-only so unit tests error if this is committed.

**Comment by aurelien-reeves** [comment]
Shouldn't we deprecate it for at least one entire major version before removing it?

**Comment by davidjgoss** [comment]
> Shouldn't we deprecate it for at least one entire major version before removing it?

It's been kinda pseudo-deprecated since we had async/await natively and then `BeforeStep` and `AfterStep` in 7.0.0. I'd be really keen to remove now as it's responsible for a chunk of complexity and there's no case for using it any more (unlike e.g. `--no-strict` which might be more subjective). We could do a patch release on 7.x.x with an explicit warning?

**Comment by aurelien-reeves** [comment]
> 
> It's been kinda pseudo-deprecated since we had async/await natively and then `BeforeStep` and `AfterStep` in 7.0.0.

We need to let our users know about that. They may not be aware of that.

> I'd be really keen to remove now as it's responsible for a chunk of complexity and there's no case for using it any more

Yes, of course, I totally agree with that. We have to remove it ASAP if it is useless complex noise in the code base.
But our users have to be informed first to let them some time to update their own code. If we are too quickly into removing it, some users may pin cucumber to v7 for a while, then forget about it, then keep that version for years - worst case scenario 😅

> We could do a patch release on 7.x.x with an explicit warning?

Yes, that would be a good thing 👍

**Review by davidjgoss — APPROVED** [review]
LGTM

Going to do a 7.3.1 via https://github.com/cucumber/cucumber-js/pull/1731 to give more explicit deprecation message.

**Comment by aslakhellesoy** [comment]
Not super related, but if you do a 7.3.1, could you please upgrade as many `@cucumber/*` dependencies as possible at the same time? There have been a few bug fixes recently since 7.3.0.

**Comment by davidjgoss** [comment]
@aslakhellesoy I'll have a look, although if updates would require the major version change on messages for the relocation of `willBeRetried` then that will have to wait for 8.0.0.

**Comment by aurelien-reeves** [comment]
This has been merged, but 7.3.1 with deprecation notice has not been released yet.

Should we undo the merge?

@charlierudolph @davidjgoss

**Comment by davidjgoss** [comment]
@aurelien-reeves im doing the release today

**Comment by aurelien-reeves** [comment]
Which one? 7.3.1 without that PR? Or 8.0.0 with it?

**Comment by davidjgoss** [comment]
Just 7.3.1 - there's quite a [bit more to do](https://github.com/cucumber/cucumber-js/milestone/12) for 8.0.0 yet

**Pull request merged**